### PR TITLE
Update template infrastructure issues

### PR DESCRIPTION
**Pull request by AlitzelMendez**
We want to use the template more broadly (reporting any infra issue) and having specific instructions when we have a known build error. 

### To double check:

* [x] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 1]
```diff
@@ -9,22 +9,28 @@ body:
...
     id: build-leg
     attributes:
       label: Build leg reported
-      description: If the issue was opened from the build analysis, this was automatically fill, feel free to update it with any impacted leg.
+      description: Enter the name of the stage/job where the build break happened.
   - type: textarea
     id: error-message
     attributes:
-      label: Error message
+      label: Action requiered for the engineering services team
```
Comment (missymessa) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:23]:
```suggestion
      label: Action required for the engineering services team
```

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 2]
```diff
@@ -9,22 +9,28 @@ body:
...
-      description: If the issue was opened from the build analysis, this was automatically fill, feel free to update it with a link to any impacted build.
+      description: Enter link to failing build, e.g. https://dev.azure.com/{organization}/{project}/_build/results?buildId={buildId} 
     validations:
       required: true
   - type: input
     id: build-leg
     attributes:
       label: Build leg reported
-      description: If the issue was opened from the build analysis, this was automatically fill, feel free to update it with any impacted leg.
+      description: Enter the name of the stage/job where the build break happened.
```
Comment (missymessa) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:19]:
```suggestion
      description: Enter the name of the stage/job where the build failed.
```

**Comment by missymessa** [comment]
Mentioned this PR in the team channel so we could get more eyes on it from folks who aren't familiar with what this is for.

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 3]
```diff
@@ -9,22 +9,28 @@ body:
     id: build
     attributes:
       label: Build
-      description: If the issue was opened from the build analysis, this was automatically fill, feel free to update it with a link to any impacted build.
+      description: Enter link to failing build, e.g. https://dev.azure.com/{organization}/{project}/_build/results?buildId={buildId}
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:12]: The template mentions this is for issues across multiple builds. Does this expect multiple examples or just one of them?

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 4]
```diff
@@ -9,22 +9,28 @@ body:
...
     attributes:
       label: Build leg reported
-      description: If the issue was opened from the build analysis, this was automatically fill, feel free to update it with any impacted leg.
+      description: Enter the name of the stage/job where the build failed.
   - type: textarea
     id: error-message
     attributes:
-      label: Error message
+      label: Action required for the engineering services team
       description: This information is going to be fill in by the engineering services team.
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:24]:
```suggestion
      description: This information is going to be filled in by the engineering services team. 
```

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 5]
```diff
@@ -9,22 +9,28 @@ body:
...
+      label: Action required for the engineering services team
       description: This information is going to be fill in by the engineering services team. 
       value: |
-         The engineering services team will fill in the error message using [known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section).
-         
+        To triage this issue:
+        - [ ] Open the failing build above and investigate
+        - [ ] Add a comment explaining your findings 
+
+        If this is an issue that is causing build breaks across multiple builds and would get benefit from be listed on the [build analysis check](https://github.com/dotnet/arcade/blob/174acbcafeba3ae339c1935ec48812514ee09837/Documentation/Projects/Build%20Analysis/Introduction.md), follow the next steps:
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:30]: This already says that it should've only been used for issues across multiple builds, so this conditional seems out of place.

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
-name: Report Infrastructure Issue - Build Break
-description: File an infrastructure issue causing build breaks to be investigated by the engineering services team.
-labels: ["First Responder", "Detected By - Customer", "Known Build Error"]
+name: Report an infrastructure issue
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:1]: Maybe also include that this is an infrastructure issue that is seen in multiple builds in the title? People sometimes only read titles

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
+description: Report an infrastructure issue causing build breaks across multiple builds
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:2]: [nit] the wording seems a bit redundant, but I'm not sure I even like this suggestion
```suggestion
description: Report an infrastructure issue causing failures across multiple builds
```

**Review by riarenas — COMMENTED** [review]
Some comments on the wording. I started reading the template as something that should only be used for issues seen in multiple builds, but ended up wondering if that was the case by the end.

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 8]
```diff
@@ -9,22 +9,28 @@ body:
     id: build
     attributes:
       label: Build
-      description: If the issue was opened from the build analysis, this was automatically fill, feel free to update it with a link to any impacted build.
+      description: Enter link to at least one failing build, e.g. https://dev.azure.com/{organization}/{project}/_build/results?buildId={buildId}
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:12]: Input fields are one line only, what's the expected format if someone is to post multiple builds? Should it be a textarea instead?

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 9]
```diff
@@ -9,22 +9,28 @@ body:
...
+      label: Action required for the engineering services team
+      description: This information is going to be filled in by the engineering services team. 
       value: |
-         The engineering services team will fill in the error message using [known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section).
-         
+        To triage this issue:
+        - [ ] Open the failing build above and investigate
+        - [ ] Add a comment explaining your findings 
+
+        If this is an issue that is causing build breaks across multiple builds and would get benefit from be listed on the [build analysis check](https://github.com/dotnet/arcade/blob/174acbcafeba3ae339c1935ec48812514ee09837/Documentation/Projects/Build%20Analysis/Introduction.md), follow the next steps:
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:30]:
```suggestion
        If this is an issue that is causing build breaks across multiple builds and would get benefit from being listed on the [build analysis check](https://github.com/dotnet/arcade/blob/174acbcafeba3ae339c1935ec48812514ee09837/Documentation/Projects/Build%20Analysis/Introduction.md), follow the next steps:
```

File: .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml [hunk 10]
```diff
@@ -9,22 +9,28 @@ body:
...
+        1. Add the label "Known Build Error"
+        2. Edit this issue and add an error string in the Json below that can help us match this issue with future build breaks. You can use the [known issues documentation](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section)
```
Comment (riarenas) [inline .github/ISSUE_TEMPLATE/build-break-infrastructure-issue-template.yml:32]: [nit] 
```suggestion
        2. Edit this issue and add an error string in the Json below that can help us match this issue with future build breaks. You should use the [known issues documentation](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section)
```

**Review by riarenas — COMMENTED** [review]
My initial reaction to using this issue for all infrastructure failures was meh, as my understanding was that this would only be used by automation in the dev workflow things, but I don't have a concrete reason why I don't think it could work, so let's give it a try?

**Review by riarenas — APPROVED** [review]

**Review by riarenas — DISMISSED** [review]

**Review by ulisesh — APPROVED** [review]
LGTM

**Comment by riarenas** [comment]
Just making sure @ilyas1974  is aware that any time anyone opens an issue using this template, even if they just think it's an infra error, it will create a mention and start the countdown for FR to meet the SLA.

**Review by ulisesh — APPROVED** [review]

**Pull request merged**